signing/src/main/java/tech/pegasys/web3signer/signing/secp256k1/aws/AwsKMSSigner.java

```diff
@@ -0,0 +1,110 @@
+import tech.pegasys.web3signer.signing.secp256k1.Signer;
+import tech.pegasys.web3signer.signing.secp256k1.util.Eth1SignatureUtil;
+
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.interfaces.ECPublicKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.X509EncodedKeySpec;
+
+import org.web3j.crypto.Hash;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.core.SdkBytes;
+import software.amazon.awssdk.services.kms.KmsClient;
+import software.amazon.awssdk.services.kms.model.GetPublicKeyRequest;
+import software.amazon.awssdk.services.kms.model.GetPublicKeyResponse;
+import software.amazon.awssdk.services.kms.model.KeySpec;
+import software.amazon.awssdk.services.kms.model.MessageType;
+import software.amazon.awssdk.services.kms.model.SignRequest;
+import software.amazon.awssdk.services.kms.model.SigningAlgorithmSpec;
+
+public class AwsKMSSigner implements Signer {
+  private final AwsKMSMetadata awsKMSMetadata;
+  private final AwsCredentialsProvider awsCredentialsProvider;
+  private final ECPublicKey ecPublicKey;
+  // required for eth1 signing. Filecoin signing doesn't need it.
+  private final boolean applySha3Hash;
+
+  public AwsKMSSigner(final AwsKMSMetadata awsKMSMetadata, boolean applySha3Hash) {
+    this.awsKMSMetadata = awsKMSMetadata;
+    this.applySha3Hash = applySha3Hash;
+    awsCredentialsProvider =
+        AwsCredentialsProviderFactory.createAwsCredentialsProvider(
+            awsKMSMetadata.getAuthenticationMode(), awsKMSMetadata.getAwsCredentials());
+    try (final KmsClient kmsClient =
+        AwsKMSClientFactory.createKMSClient(
+            awsCredentialsProvider,
+            awsKMSMetadata.getRegion(),
+            awsKMSMetadata.getEndpointOverride())) {
+      this.ecPublicKey = getECPublicKey(kmsClient, awsKMSMetadata.getKmsKeyId());
+    }
+  }
+
+  public static ECPublicKey getECPublicKey(final KmsClient kmsClient, final String kmsKeyId) {
+    // kmsClient can be null/closed if close method has been called.
+    checkArgument(kmsClient != null, "KmsClient is not initialized");
+
+    final GetPublicKeyRequest getPublicKeyRequest =
+        GetPublicKeyRequest.builder().keyId(kmsKeyId).build();
+    final GetPublicKeyResponse publicKeyResponse = kmsClient.getPublicKey(getPublicKeyRequest);
+    KeySpec keySpec = publicKeyResponse.keySpec();
+    if (keySpec != KeySpec.ECC_SECG_P256_K1) {
+      throw new RuntimeException("Unsupported key spec from AWS KMS: " + keySpec.toString());
+    }
+
+    final X509EncodedKeySpec encodedKeySpec =
+        new X509EncodedKeySpec(publicKeyResponse.publicKey().asByteArray());
+    try {
+      KeyFactory keyFactory = KeyFactory.getInstance("EC");
+      return (ECPublicKey) keyFactory.generatePublic(encodedKeySpec);
+    } catch (NoSuchAlgorithmException | InvalidKeySpecException e) {
+      throw new RuntimeException(e);
```

Comment:
Are the error states tested somewhere?

Reply:
This is very unlikely to happen error. I've updated the code to add a comment explaining as such. Since these are checked exceptions, I am wrapping them in a runtime exception. Additionally, I've now used Bouncycastle provider instead of using the default Sun implementation.